Menus/PauseMenuGUI.cpp

```diff
@@ -0,0 +1,325 @@
+				m_PauseMenuButtons[pauseMenuButton]->MoveRelative(0, yOffset * (disableSaving ? -1 : 1));
+			}
+			m_PauseMenuBox->MoveRelative(0, yOffset / 2 * (disableSaving ? 1 : -1));
+		}
+		m_SavingButtonsDisabled = disableSaving;
+
+		if (m_ModManagerButtonDisabled != disableModManager) {
+			GUIButton *modManagerButton = m_PauseMenuButtons[PauseMenuButton::ModManagerButton];
+
+			modManagerButton->SetEnabled(!disableModManager);
+			modManagerButton->SetVisible(!disableModManager);
+
+			int yOffset = m_PauseMenuButtons[PauseMenuButton::ModManagerButton]->GetHeight();
+
+			m_PauseMenuButtons[PauseMenuButton::ResumeButton]->MoveRelative(0, yOffset * (disableModManager ? -1 : 1));
+			m_PauseMenuBox->MoveRelative(0, yOffset / 2 * (disableSaving ? 1 : -1));
+		}
+		m_ModManagerButtonDisabled = disableModManager;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PauseMenuGUI::SetActiveMenuScreen(PauseMenuScreen screenToShow, bool playButtonPressSound) {
+		if (screenToShow != m_ActiveMenuScreen) {
+			m_ActiveMenuScreen = screenToShow;
+			if (playButtonPressSound) {
+				g_GUISound.ButtonPressSound()->Play();
+			}
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PauseMenuGUI::StoreFrameForUseAsBackdrop() {
+		BITMAP *backbuffer = g_FrameMan.GetBackBuffer32();
+		blit(backbuffer, m_BackdropBitmap, 0, 0, 0, 0, backbuffer->w, backbuffer->h);
+
+		set_trans_blender(96, 96, 96, 96);
+		draw_trans_sprite(m_BackdropBitmap, g_FrameMan.GetOverlayBitmap32(), 0, 0);
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	PauseMenuGUI::PauseMenuUpdateResult PauseMenuGUI::Update() {
+		m_UpdateResult = PauseMenuUpdateResult::NoEvent;
+
+		if (g_ConsoleMan.IsEnabled() && !g_ConsoleMan.IsReadOnly()) {
+			return m_UpdateResult;
+		}
+
+		bool backToMainScreen = false;
+
+		switch (m_ActiveMenuScreen) {
+			case PauseMenuScreen::MainScreen:
+				backToMainScreen = HandleInputEvents();
+				BlinkResumeButton();
+				break;
+			case PauseMenuScreen::SettingsScreen:
+				backToMainScreen = m_SettingsMenu->HandleInputEvents();
+				m_ActiveDialogBox = m_SettingsMenu->GetActiveDialogBox();
+				break;
+			case PauseMenuScreen::ModManagerScreen:
+				backToMainScreen = m_ModManagerMenu->HandleInputEvents();
+				break;
+			default:
+				break;
+		}
+		HandleBackNavigation(backToMainScreen);
+
+		return m_UpdateResult;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PauseMenuGUI::HandleBackNavigation(bool backButtonPressed) {
+		if (!m_ActiveDialogBox && (backButtonPressed || g_UInputMan.KeyPressed(SDLK_ESCAPE))) {
```

Comment:
It probably makes more sense to check for start press or whatever, to deal with controllers?